packages/clinicalViewLambda/src/handler.ts

```diff
@@ -2,32 +2,64 @@ import {SpineClient} from "@nhsdigital/eps-spine-client/lib/spine-client"
 import {LogLevel} from "@aws-lambda-powertools/logger/types"
 import {Logger} from "@aws-lambda-powertools/logger"
 import {createSpineClient} from "@nhsdigital/eps-spine-client"
-import {APIGatewayEvent} from "aws-lambda"
+import {APIGatewayEvent, APIGatewayProxyEventHeaders, APIGatewayProxyEventQueryStringParameters} from "aws-lambda"
 import middy from "@middy/core"
 import {ClinicalViewParams} from "@nhsdigital/eps-spine-client/lib/live-spine-client"
+import {DOMParser} from "xmldom"
+import {AxiosResponse} from "axios"
 
 const LOG_LEVEL = process.env.LOG_LEVEL as LogLevel
-export const logger = new Logger({serviceName: "clinicalViewLambda", logLevel: LOG_LEVEL})
-const spineClient = createSpineClient(logger)
+export const defaultLogger = new Logger({serviceName: "clinicalViewLambda", logLevel: LOG_LEVEL})
+const defaultSpineClient = createSpineClient(defaultLogger)
 
 type HandlerParams = {
     logger: Logger,
     spineClient: SpineClient
 }
 
-export const apiGatewayHandler = async (params: HandlerParams, event: APIGatewayEvent) => {
+type HandlerResponse = {
+    data: {
+        prescriptionId: string,
+        prescriptionStatus?: string,
+        error?: string
+    },
+    status: number
+}
+
+export const apiGatewayHandler = async (params: HandlerParams, event: APIGatewayEvent): Promise<HandlerResponse> => {
+  const logger = params.logger
+
   const inboundHeaders = event.headers
+  const queryStringParameters = event.queryStringParameters ?? {}
+  const prescriptionId = event.queryStringParameters?.prescriptionId ?? ""
+
+  const clinicalViewParams = buildClinicalViewParams(inboundHeaders, queryStringParameters)
+
+  let spineResponse
+  try {
+    spineResponse = await params.spineClient.clinicalView(inboundHeaders, clinicalViewParams)
+  } catch (error) {
```

Comment:
why not just use the error handler middleware?

Reply:
db075b846b1def9acd29e9b0a076fbdd261903c5